api/capi/src/nnstreamer-capi-single-new.c

```diff
@@ -198,17 +198,86 @@ invoke_thread (void *arg)
   return NULL;
 }
 
+/**
+ * @brief Gets the tensors info from tensor-filter.
+ */
+static void
+ml_single_get_tensors_info_from_filter (GTensorFilterSingle * filter, gboolean is_input,
+    ml_tensors_info_h * info)
+{
+  ml_tensors_info_s *input_info;
+  GstTensorsInfo gst_info;
+  gchar *val;
+  guint rank;
+  gchar *prop_prefix, *prop_name;
+
+  if (is_input)
+    prop_prefix = g_strdup ("input");
+  else
+    prop_prefix = g_strdup ("output");
+
+  /* allocate handle for tensors info */
+  ml_tensors_info_create (info);
+  input_info = (ml_tensors_info_s *) (*info);
+
+  ml_tensors_info_initialize (input_info);
+  gst_tensors_info_init (&gst_info);
+
+  /* get dimensions */
+  prop_name = g_strdup_printf ("%s", prop_prefix);
+  g_object_get (filter, prop_name, &val, NULL);
```

Comment:
property_name of g_object_get is const char *; thus, you don't need strdup. You can even directly give "input" here (or static const gchar *)

Reply:
updated.